### PR TITLE
fix(trace.py): recursively remove ignored keys

### PR DESCRIPTION
**Pull request by ranrib**

### CODE REVIEW

File: epsagon/trace.py [hunk 1]
```diff
@@ -953,6 +953,15 @@ def remove_ignored_keys(self, input_dict):
...
                 if isinstance(value, dict):
                     copied_dict[key] = self.remove_ignored_keys(value)
+                else:
+                    try:
+                        json_value = json.loads(value)
+                        if isinstance(json_value, dict):
+                            copied_dict[key] = (json.dumps(
+                                self.remove_ignored_keys(json_value)
+                            ))
+                    except (TypeError, json.errors.JSONDecodeError):
```
Comment (maorlx) [inline epsagon/trace.py:963]: when can TypeError be raised?

File: epsagon/trace.py [hunk 2]
```diff
@@ -953,6 +953,15 @@ def remove_ignored_keys(self, input_dict):
                 value = input_dict[key]
                 if isinstance(value, dict):
                     copied_dict[key] = self.remove_ignored_keys(value)
+                else:
+                    try:
+                        json_value = json.loads(value)
```
Comment (maorlx) [inline epsagon/trace.py:958]: Consider checking if value is str before calling json.loads

File: epsagon/trace.py (hunk 1 above)
Comment (ranrib) [reply epsagon/trace.py:963]: Depends on the Python version you'll get either one of these

File: epsagon/trace.py (hunk 2 above)
Comment (ranrib) [reply epsagon/trace.py]: Right 👍

**Pull request merged**